File: fsharp/sum-of-multiples/SumOfMultiples.fs
module SumOfMultiples

let isMultipleOf x y = y <> 0 && x % y = 0
let isMultipleOfAny list x = List.exists (isMultipleOf x) list

let sum (numbers: int list) (upperBound: int): int =
    seq { 1 .. upperBound - 1 }
    |> Seq.filter (isMultipleOfAny numbers)
    |> Seq.sum

File: kotlin/sum-of-multiples/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {

    fun sum(factors: Set<Int>, limit: Int): Int =
        (1 until limit)
            .filter { factors.any { factor -> it.isMultipleOf(factor) } }
            .sum()

    private fun Int.isMultipleOf(x: Int) = when (x) {
        0 -> false
        else -> this % x == 0
    }
}
